src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -665,19 +746,81 @@ private void ImportCall(ILOpcode opcode, int token)
                 return;
             }
 
+            if (opcode == ILOpcode.newobj)
+            {
+                if (callee.OwningType.IsString)
+                {
+                    // String constructors actually look like regular method calls
+                    IMethodNode node = _compilation.NodeFactory.StringAllocator(callee);
+                    _dependencies.Add(node);
+                    callee = node.Method;
+                    opcode = ILOpcode.call;
+                }
+                else
+                {
+                    AllocateObject(callee.OwningType);
+                    ImportDup();
+                }
+            }
+
             // we don't really have virtual call support, but we'll treat it as direct for now
-            if (opcode != ILOpcode.call && opcode !=  ILOpcode.callvirt)
+            if (opcode != ILOpcode.call && opcode !=  ILOpcode.callvirt && opcode != ILOpcode.newobj)
             {
                 throw new NotImplementedException();
             }
             if (opcode == ILOpcode.callvirt && callee.IsAbstract)
             {
                 throw new NotImplementedException();
             }
-
+            AddMethodReference(callee);
             HandleCall(callee);
         }
 
+        private void AllocateObject(TypeDesc type)
+        {
+            MetadataType metadataType = (MetadataType)type;
+            int objectSize = metadataType.InstanceByteCount.AsInt;
+            LLVMValueRef objectSizeValue = BuildConstInt32(objectSize);
+            LLVMValueRef allocatedMemory = LLVM.BuildMalloc(_builder, LLVM.ArrayType(LLVM.Int8Type(), (uint)objectSize), "newobj");
+            LLVMValueRef castMemory = LLVM.BuildPointerCast(_builder, allocatedMemory, LLVM.PointerType(LLVM.Int8Type(), 0), "castnewobj");
+            if (MemsetFunction.Pointer == IntPtr.Zero)
+            {
+                MemsetFunction = LLVM.AddFunction(Module, "llvm.memset.p0i8.i32", LLVM.FunctionType(LLVM.VoidType(), new LLVMTypeRef[] { LLVM.PointerType(LLVM.Int8Type(), 0), LLVM.Int8Type(), LLVM.Int32Type(), LLVM.Int32Type(), LLVM.Int1Type() }, false));
+            }
+            LLVM.BuildCall(_builder, MemsetFunction, new LLVMValueRef[] { castMemory, BuildConstInt8(0), objectSizeValue, BuildConstInt32(1), BuildConstInt1(0) }, String.Empty);
```

Comment:
If you want, we could only zero memory after the eeTypePointer (Hopefully the LLVM optimizer handles this, but I haven't tested)

Reply:
Given this code is only temporary, until the runtime is built, I'm inclined to just leave it.